tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/websocket/FailoverClientTest.java

```diff
@@ -0,0 +1,152 @@
+        closeConnection(client, remoteServer15300);
+    }
+
+    @Test(description = "Tests the failover webSocket client by doesn't start the any server in the targets URLs")
+    public void testFailingFailover() throws URISyntaxException, InterruptedException {
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        WebSocketTestClient client = initiateClient(url);
+        client.setCountDownLatch(countDownLatch);
+        countDownLatch.await(TIMEOUT_IN_SECS, TimeUnit.SECONDS);
+        CloseWebSocketFrame closeWebSocketFrame = client.getReceivedCloseFrame();
+        Assert.assertNotNull(closeWebSocketFrame);
+        Assert.assertEquals(closeWebSocketFrame.statusCode(), 1011);
+        Assert.assertTrue(closeWebSocketFrame.reasonText().contains("Unexpected condition"));
+        closeWebSocketFrame.release();
+    }
+
+    @Test(description = "Tests the failover webSocket client by starts the both server in the target URLs.")
+    public void testFailoverWithBothServer() throws URISyntaxException, InterruptedException,
+            BallerinaTestException {
+        remoteServer15200 = initiateServer(15200);
+        remoteServer15300 = initiateServer(port);
+        WebSocketTestClient client = initiateClient(url);
+        sendTextDataAndAssert(client, message);
+        remoteServer15300.stop();
+        sendTextDataAndAssert(client, text);
+        sendBinaryDataAndAssert(client);
+        closeConnection(client, remoteServer15200);
+    }
+
+    @Test(description = "Tests the failover webSocket client by starts the both server in the target URLs.")
+    public void testFailoverWithBothServerFirstStartSecondServer() throws URISyntaxException, InterruptedException,
+            BallerinaTestException {
+        remoteServer15200 = initiateServer(15200);
+        WebSocketTestClient client = initiateClient(url);
+        sendTextDataAndAssert(client, message);
+        remoteServer15300 = initiateServer(port);
+        remoteServer15200.stop();
+        sendTextDataAndAssert(client, text);
+        closeConnection(client, remoteServer15300);
+    }
+
+    @Test(description = "Tests handshake's waiting time exception")
+    public void testCountDownLatch() throws URISyntaxException, InterruptedException, BallerinaTestException {
+        WebSocketTestClient serverClient = initiateClient("ws://localhost:21033/basic/ws");
+        remoteServer15300 = initiateServer(port);
+        WebSocketTestClient client = initiateClient("ws://localhost:21034");
+        CountDownLatch countDownLatch1 = new CountDownLatch(1);
+        countDownLatch1.await(8, TimeUnit.SECONDS);
+        sendTextDataAndAssert(client, message);
+        remoteServer15300.stop();
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        countDownLatch.await(8, TimeUnit.SECONDS);
+        Assert.assertEquals(client.getTextReceived(), "error {ballerina/http}WsGenericError " +
+                "message={ballerina/http}WsGenericError");
```

Comment:
Shall we fix the relevant error message? It doesn't give an explanation of the failure. check if we need to create a Hanshake related error instead of a generic error?